great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1144,12 +1144,39 @@ def _build_selectable_from_batch_spec(self, batch_spec: BatchSpec) -> sqlalchemy
 
         return sa.select("*").select_from(selectable).where(partition_clause)  # type: ignore[arg-type] # FIXME CoP
 
+
     def _subselectable(self, batch_spec: BatchSpec) -> sqlalchemy.Selectable:
         table_name = batch_spec.get("table_name")
         query = batch_spec.get("query")
         selectable: sqlalchemy.Selectable
         if table_name:
-            selectable = sa.table(table_name, schema=batch_spec.get("schema_name", None))
+            # Handle Databricks identifier quoting if needed
+            if self.dialect_name == GXSqlDialect.DATABRICKS:
+                import re
+                schema_name = batch_spec.get("schema_name", None)
+                table_name_str = str(table_name)
+                schema_name_str = str(schema_name) if schema_name else None
+
+                # Check if the table name needs special escaping for Databricks
+                if re.match(r"^\d", table_name_str) or re.search(r"[.\s\-#@]", table_name_str):
+                    # For Databricks, create a subquery with backticks
+                    clean_table_name = table_name_str.strip('"').strip("'").strip("`")
+                    if schema_name_str:
+                        clean_schema_name = schema_name_str.strip('"').strip("'").strip("`")
+                        # Use a select with text to control the identifier quoting
+                        selectable = sa.select(sa.text("*")).select_from(
+                            sa.text(f"`{clean_schema_name}`.`{clean_table_name}`")
+                        ).subquery()
+                    else:
+                        selectable = sa.select(sa.text("*")).select_from(
+                            sa.text(f"`{clean_table_name}`")
+                        ).subquery()
+                else:
+                    # Standard table that doesn't need special escaping
+                    selectable = sa.table(table_name, schema=schema_name)
+            else:
+                # Standard behavior for all other databases
+                selectable = sa.table(table_name, schema=batch_spec.get("schema_name", None))
```

Comment:
Instead of adding a new condition here, we would rather keep this logic [in this table_name validator](https://github.com/great-expectations/great_expectations/blob/967c2b669960150b3370ba5a176dac0c0493a65e/great_expectations/datasource/fluent/databricks_sql_datasource.py#L141). We already have logic there for handling "quoted" table names. Feel free to rename that method if it makes more sense to (a validator gets called automatically based on the decorator string). Hopefully, it will be easier to manipulate the table name string directly, rather than work with all of these SQL Alchemy constructs.